### PR TITLE
ci: install golang if not currently installed from ci tasks

### PR DESCRIPTION
**Pull request by justinsb**

### CODE REVIEW

File: dev/tasks/install-tools [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+cd "${REPO_ROOT}"
+
+if [[ ! -x go ]]; then
+  echo "Must install go; assuming we are running in a container"
+  curl -L -o /tmp/go.tar.gz https://go.dev/dl/go1.22.4.linux-amd64.tar.gz
```
Comment (yuwenma) [inline dev/tasks/install-tools:25]: I used to hit a controller-gen problem in go1.22 https://github.com/kubernetes-sigs/controller-tools/issues/880
If installing golang, can we install go1.23 instead?

Comment (justinsb) [reply dev/tasks/install-tools:25]: I think 1.22.4 is the latest: https://go.dev/doc/install

I think we updated controller-gen so we should be good with 1.22?

Comment (yuwenma) [reply dev/tasks/install-tools:25]: you are absolutely right!

**Pull request merged**